internal/server/api_snapshots.go

```diff
@@ -79,3 +84,86 @@ func convertSnapshotManifest(m *snapshot.Manifest) *serverapi.Snapshot {
 
 	return e
 }
+
+func (s *Server) handleSnapshotDelete(ctx context.Context, r *http.Request, body []byte) (interface{}, *apiError) {
+	oidstr := mux.Vars(r)["snapshotID"]
+	rep := s.rep
+
+	oid, err := object.ParseID(oidstr)
+	if err != nil {
+		log(ctx).Errorf("error parsing object id", err)
+		return nil, requestError(serverapi.ErrorMalformedRequest, "invalid object id")
+	}
+
+	// logic here should eventually be similar to that in commandSnapshotDelete run
+	// for now just assume the oid is a manifest ID and handle simple not found errors
+	_, err = snapshot.LoadSnapshot(ctx, rep, manifest.ID(oid))
+	if err != nil {
+		log(ctx).Errorf("error loading snapshot %v", oid)
+		return nil, requestError(serverapi.ErrorNotFound, "snapshot ID not found")
+	}
+
+	log(ctx).Debugw("would delete snapshot with id", "id", oid)
+
+	resp := serverapi.DeletionResponseSingle{
+		Type:    "snapshot",
+		Deleted: false,
+		Error:   "deletion not yet fully implemented",
+	}
+
+	return resp, nil
+}
+
+func (s *Server) handleSnapshotDeleteBatch(ctx context.Context, r *http.Request, body []byte) (interface{}, *apiError) {
+	var req serverapi.BatchSnapshotDeleteRequest
+
+	rep := s.rep
+
+	if err := json.Unmarshal(body, &req); err != nil {
+		return nil, requestError(serverapi.ErrorMalformedRequest, "malformed request body")
+	}
+
+	if len(req.Snapshots) == 0 {
+		return nil, requestError(serverapi.ErrorMalformedRequest, "missing list of snapshots")
+	}
+
+	resp := &serverapi.DeletionResponseBatch{
+		Items: []*serverapi.DeletionStatus{},
+	}
+
+	for _, idstr := range req.Snapshots {
+		item := &serverapi.DeletionStatus{
+			Type: "snapshot",
+			ID:   idstr,
+		}
+
+		id, err := object.ParseID(idstr)
```

Comment:
I think deletion is relatively simple, we may be better off not introducing per-item responses

if you parse all IDs early on and fail on any errors, you can simply return `Empty` on success.